feat: Add support for @<tag> in plugin name

SirWrexes opened the pull request:
Choosing a specific release of a plugin should be done in a user
friendly way. Therefore, it should be as simple as adding a tag
at the end of the plugin name consisting of username/repository.
This small modifications makes this possible using '@' as a delimiter.

Example:
    `fundle plugin 'joseluisq/gitnow@2.7.0'`

Not selecting a tag still acts the same as usual.

SirWrexes committed:
feat: Add support for @<tag> in plugin name

Choosing a specific release of a plugin should be done in a user
friendly way. Therefore, it should be as simple as adding a tag
at the end of the plugin name consisting of username/repository.
This small modifications makes this possible using '@' as a delimiter.

Example:
    `fundle plugin 'joseluisq/gitnow@2.7.0'`

Not selecting a tag still acts the same as usual.

SirWrexes commented on functions/fundle.fish:
```diff
@@ -380,6 +385,7 @@ function __fundle_plugin -d "add plugin to fundle" -a name
 		end
 	end
 	test -z "$plugin_url"; and set plugin_url (__fundle_get_url $name)
+    set name (echo "$name" | cut -d'@' -f1)
```
I use spaces for indentation and didn't notice you were using tabs.
I'll fix that and squash the commits.

jorgebucaran replied:
Here as well ([comment](https://github.com/danhper/fundle/pull/54/files#r588839957)).

SirWrexes commented:
I realise I should probably edit the `README.md`. :thinking: 
I'll do that.

SirWrexes commented on README.md:
````diff
@@ -89,6 +89,12 @@ fundle plugin 'danhper/fish-fastdir'
+```
+fundle plugin 'joseluisq/gitnow@2.7.0'
+```
+will install Gitnow release 2.7.0 at https://github.com/joseluisq/gitnow/releases/tag/2.7.0.
````
I know this is not the repo you chose as your example though the whole file, but it's the first that came to mind that had versioning and thus fitting this example.

jorgebucaran replied:
Or ilancosman/tide@v4.1.1 or jorgebucaran/nvm.fish@2.1.0.

jorgebucaran commented on functions/fundle.fish:
```diff
@@ -123,7 +123,12 @@ function __fundle_check_date -d "check date"
 
 function __fundle_get_url -d "returns the url for the given plugin" -a repo
-	echo "https://github.com/$repo.git"
+    set tag  (echo "$repo" | cut -s -d'@' -f2)
+    set repo (echo "$repo" | cut -d'@' -f1)
+    set url "https://github.com/$repo.git"
```
Faster and fishier would be to use the delightful [`string`](https://fishshell.com/docs/current/cmds/string.html) builtin.

SirWrexes replied:
You're right ! I'll change that.

SirWrexes commented:
I have made the changes using the `string split` function, squashed the commits and pushed.
Also I added a small note in the description of the `--url` option explaining it overrides tag selection via `@` (since it manually sets the URL for the repo).

danhper commented:
@SirWrexes Again, very sorry for not noticing this and thanks a lot!
